core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProviderSpec.scala

```diff
@@ -128,5 +135,116 @@ class LocalEntityProviderSpec extends AnyWordSpecLike with Matchers with TestDri
+      attrNamesCache shouldBe Map.empty
+
+      //if there's nothing in the cache, we can assert that the results we're getting below are _not_ from the cache
+      entityTypeMetadataResult should contain theSameElementsAs expectedResultWhenUsingFullQueries
+    }
+
+    "not use cache for entityTypeMetadata when useCache=false even if cache is up to date and cache is enabled" in withLocalEntityProviderTestDatabase { dataSource =>
+      val workspaceContext = runAndWait(dataSource.dataAccess.workspaceQuery.findById(localEntityProviderTestData.workspace.workspaceId)).get
+      val localEntityProvider = new LocalEntityProvider(workspaceContext, slickDataSource, cacheEnabled = true)
+
+      //Update the entityCacheLastUpdated field to be identical to lastModified, so we can test our scenario of having a fresh cache
+      runAndWait(workspaceQuery.updateCacheLastUpdated(workspaceContext.workspaceIdAsUUID, new Timestamp(workspaceContext.lastModified.getMillis)))
+
+      val entityTypeMetadataResult = runAndWait(DBIO.from(localEntityProvider.entityTypeMetadata(useCache = false)))
+
+      val typeCountCache = runAndWait(dataSource.dataAccess.entityTypeStatisticsQuery.getAll(workspaceContext.workspaceIdAsUUID))
+      val attrNamesCache = runAndWait(dataSource.dataAccess.entityAttributeStatisticsQuery.getAll(workspaceContext.workspaceIdAsUUID))
+
+      //assert that there is something in the cache for this workspace
+      typeCountCache should not be Map.empty
+      attrNamesCache should not be Map.empty
+
+      entityTypeMetadataResult should contain theSameElementsAs expectedResultWhenUsingFullQueries
+    }
+
+    "not use the cache when it's disabled even if cache is up to date and useCache=true" in withLocalEntityProviderTestDatabase { dataSource =>
+      val workspaceContext = runAndWait(dataSource.dataAccess.workspaceQuery.findById(localEntityProviderTestData.workspace.workspaceId)).get
+      val localEntityProvider = new LocalEntityProvider(workspaceContext, slickDataSource, false)
+
+      //Update the entityCacheLastUpdated field to be identical to lastModified, so we can test our scenario of having a fresh cache
+      runAndWait(workspaceQuery.updateCacheLastUpdated(workspaceContext.workspaceIdAsUUID, new Timestamp(workspaceContext.lastModified.getMillis)))
+
+      val entityTypeMetadataResult = runAndWait(DBIO.from(localEntityProvider.entityTypeMetadata(true)))
+
+      val typeCountCache = runAndWait(dataSource.dataAccess.entityTypeStatisticsQuery.getAll(workspaceContext.workspaceIdAsUUID))
+      val attrNamesCache = runAndWait(dataSource.dataAccess.entityAttributeStatisticsQuery.getAll(workspaceContext.workspaceIdAsUUID))
+
+      //assert that there is something in the cache for this workspace
+      typeCountCache should not be Map.empty
+      attrNamesCache should not be Map.empty
+
+      entityTypeMetadataResult should contain theSameElementsAs expectedResultWhenUsingFullQueries
+    }
+
```

Comment:
another simple smoke test:
* Create workspace, populate entities
* get entity metadata (cache does not yet exist)
* populate cache
* get entity metadata, compare result to previous, should be the same.

Reply:
LocalEntityProvider doesn't handle any of the work to populate the cache but I just added a test that verifies that we update the timestamp for a workspace without an existing cache.

Reply:
my apologies, I didn't mean the test had to go in this spec!